### ISSUE TITLE
Feature Request - Callback

### ISSUE THREAD

**Issue by asoap** [hideakitai/Tween#5]
I'm looking for library that does tweening and also call backs.  This one looks soooooo close.

What I would like to do is:

```
    timeline.add(target) // target tweens
        .init(0)         // init target value (optional)
        .offset(1000)    // delay start 1000[ms] (optional)
        .then_call(some_function)
        .then(10, 5000)  // to 10 in 5000[ms]
        .then(5, 5000)   // then to 5 in 5000[ms]
        .then_call(some_function)
        .hold(1000)      // then stops 1000[ms]
        .then(0, 5000);  // then to 0 in 5000[ms]
```
I'm more interested in setting up a sequence where I can execute and change some code in the middle of a tween.  

I could possibly just fork this and give it a go.

**Comment by hideakitai** [comment]
Cool! Do you want to replace the easing with a custom callback? Do you have any example usages in other libraries? (tween.js?)

**Comment by asoap** [comment]
I mostly want it to use a timer and setup a sequence of function calls.  But I think the tweening should be kept.  I could always just change the value of a float that doesn't matter.  I say this because there is likely a situation where you want the function calls AND a tween.  

An example from another library is from Greensock.  I've been using a library from these guys for many years. 

https://greensock.com/get-started/#tweening-basics

Specifically they have the onComplete, onUpdate, and onStart attributes that you can use depending on the tween.

However if you wanted to do a sequence you would just call the library multiple times.  For example:

```
gsap.to(target, {duration: 1, x: 100});
gsap.to(target, {duration: 1, delay: 1,  x: 0, onComplete: some_function});

```
This executes the first line first, and because the second line is delayed long enough it starts right after the first is completed.

I don't think this applies to this Arduino library.  But it's how it works in the GSAP library, so I'm providing that info for clarity.

**Comment by asoap** [comment]
I have been playing around with my own code.  I'm new to Arduino but not to programming.  I got a library setup (super simple) but I figured out the call back code. 

I'm looking to inject it into this code, but this code is  doing a lot that I'm not sure I can.

I was thinking about expanding the transitions code to add a call back attribute that would point to a function to call.  I was thinking it would just be a transition of zero seconds.  

When looking at the sequence code.  I'm seeing this:

`const size_t idx = from_time_to_index(ms);`

It looks like to me that the code is figuring out where it needs to be (the index of the vector) in the sequence based on elapsed time?  If so then if I add a transition of 0 seconds it would probably just skip it.  Am I reading this correctly?

**Comment by asoap** [comment]
I added the code for a callback.  So this is how I have it setup now.  I also created a pull request.

```
        timeline.add(target) 
            .init(1)       
            .then(1, 20 * 1000, &motors_forward) 
            .then(1, 20 * 1000, &motors_backward); 
        timeline.start();
        timeline.mode(Tween::Mode::REPEAT_TL);

```

I for some reason had a weird issue when I was using Tween::Mode::REPEAT_SQ where it wouldn't call the last function.  Like it got screwed up on what function went where which seems really strange.  I also had an issue if the duration was longer than 30 seconds, the timeline would just stop.   I'm not sure if that's something I did or not though. 

This is working now in my project.
